File: tags/NHI1-0.13/theLink/example/csharp/testserver.cs
/**
 *  \file       theLink/example/csharp/testserver.cs
 *  \brief      \$Id$
 *  
 *  (C) 2009 - NHI - #1 - Project - Group
 *  
 *  \version    \$Rev$
 *  \author     EMail: aotto1968 at users.berlios.de
 *  \attention  this software has GPL permissions to copy
 *              please contact AUTHORS for additional information
 */
using System;
using csmsgque;

namespace example {

  sealed class testserver : MqS, IServerSetup {

    public testserver(MqS tmpl) : base(tmpl) {
    }

    public void GTCX () {
      SendSTART();
      SendI(LinkGetCtxId());
      SendC("+");
      if (LinkIsParent()) {
        SendI(-1);
      } else {
        SendI(LinkGetParent().LinkGetCtxId());
      }
      SendC("+");
      SendC(ConfigGetName());
      SendC(":");
      SendRETURN();
    }

    void IServerSetup.ServerSetup() {
      ServiceCreate("GTCX", GTCX);
    }

    static void Main(string[] argv) {
      testserver srv = MqFactoryS<testserver>.Add().New();
      try {
	srv.LinkCreate(argv);
	srv.ProcessEvent(MqS.WAIT.FOREVER);
      } catch (Exception ex) {
        srv.ErrorSet (ex);
      }
      srv.Exit();
    }
  }
}

File: tags/NHI1-0.13/theLink/example/perl/testserver.pl
#+
#§  \file       theLink/example/perl/testserver.pl
#§  \brief      \$Id$
#§  
#§  (C) 2009 - NHI - #1 - Project - Group
#§  
#§  \version    \$Rev$
#§  \author     EMail: aotto1968 at users.berlios.de
#§  \attention  this software has GPL permissions to copy
#§              please contact AUTHORS for additional information
#§

use strict;
use Net::PerlMsgque;

package testserver;
use base qw(Net::PerlMsgque::MqS);

  sub GTCX {
    my $ctx = shift;
    $ctx->SendSTART();
    $ctx->SendI($ctx->LinkGetCtxId());
    $ctx->SendC("+");
    if ($ctx->LinkIsParent()) {
      $ctx->SendI(-1);
    } else {
      $ctx->SendI($ctx->LinkGetParent()->LinkGetCtxId());
    }
    $ctx->SendC("+");
    $ctx->SendC($ctx->ConfigGetName());
    $ctx->SendC(":");
    $ctx->SendRETURN();
  }

  sub ServerConfig {
    my $ctx = shift;
    $ctx->ServiceCreate("GTCX",\&GTCX)
  }

  sub new {
    my $class = shift;
    my $ctx = $class->SUPER::new(@_);
    $ctx->ConfigSetServerSetup(\&ServerConfig);
    return $ctx;
  }

package main;

  our $srv = MqFactoryS::Add("testserver")->New();
  eval {
    $srv->LinkCreate(@ARGV);
    $srv->ProcessEvent(Net::PerlMsgque::WAIT_FOREVER);
  };
  if ($@) {
    $srv->ErrorSet($@);
  }
  $srv->Exit();








